src/main/java/apoc/export/util/FileUtils.java

```diff
@@ -23,6 +23,21 @@ public static CountingReader readerFor(String fileName) throws IOException {
         return new CountingReader(file);
     }
 
+    public static String changeFileUrlIfImportDirectoryConstrained(String url) throws IOException {
+        if (isFile(url) && ApocConfiguration.isEnabled("import.file.use_neo4j_config")) {
+            if (!ApocConfiguration.isEnabled("import.file.allow_read_from_filesystem"))
+                throw new RuntimeException("Import file "+url+" not enabled, please set dbms.security.allow_csv_import_from_file_urls=true in your neo4j.conf");
+            String importDir = ApocConfiguration.get("import.file.directory", null);
+            if (importDir != null && !importDir.isEmpty()) {
+                File dir = new File(importDir);
+                if (dir.exists() && dir.isDirectory() && dir.canRead()) {
+                    return "file:" + new File(importDir, url.substring(url.indexOf("/")+1)).getAbsolutePath();
```

Comment:
this is not correct yet.
- if we need to extract the path of the current URL we should use new URI(url).getPath() (after checking that it's an URL) -> try { new URI(url).getPath(); } catch(URLException ue) { return url; }
- if we *need* to construct an URI we should use the new URI("file",...,path) *not* `"file:"+...`
- in this case we don't need to construct an URL
- we should also check if the current url, already starts with the importDir, if so, we should just return it